### PR TITLE
Cleaning up the svd2rust deck.

### PR DESCRIPTION
**Pull request by jonathanpallant**
Revised the svd2rust deck.

* Talk about *bitfields* to avoid confusion with *fields in structs*
* Update link to nrf52840 PAC
* Show ZST alternative
* Note that some RISC-V chips have SVD files

### CODE REVIEW

File: training-slides/src/pac-svd2rust.md [hunk 1]
```diff
@@ -84,44 +82,82 @@ pub struct RegisterBlock {
...
+
+let uart: Uart::<0x40002000> = Uart {};
+```
+
+Note:
+
+By itself this seems a small change, but imagine a struct which represents 75
+individual peripherals. That's not impossible for a modern microcontroller.
+Holding one word for each now takes up valuable RAM!
+
```
Comment (miguelraz) [inline training-slides/src/pac-svd2rust.md:160]:
```suggestion
UART stands for Universal Asynchronous Receiver Transmitter
```

File: training-slides/src/pac-svd2rust.md [hunk 2]
```diff
@@ -134,6 +170,16 @@ graph LR
...
+Note:
+
+Although it is an Arm standard, there are examples of RISC-V based
+microcontrollers which use the same format SVD files and hence can use svd2rust.
+
+Also be aware that manufacturers often assume you will only use the SVD file to
+inspect the microcontrollers state whilst debugging, and so accuracy has been
+known to vary somewhat. Rust groups often have to maintain a set of patches to
+fix known bugs in the SVD files.
+
```
Comment (miguelraz) [inline training-slides/src/pac-svd2rust.md:182]:
```suggestion

UARTE stands for UART with EasyDMA (allows you to easily transfer data to/from RAM)
```

**Review by miguelraz — APPROVED** [review]
I left two notes to explain acronyms that I wasn't familiar with and that I didn't see defined earlier in the slide. Feel free to drop them if you don't think they are necessary.

**Comment by jonathanpallant** [comment]
Add definition of UART and UARTE in the notes.

**Comment by jonathanpallant** [comment]
Thanks!

**Pull request merged**